File: frontend/src/types/musicTrack.ts
export interface MusicTrack {
  id: number
  slug: string
  title: string
  description: string | null
  audio_url: string
  cover_url: string | null
  duration_seconds: number | null
  character_id: number | null
  character_ids: number[]
  original_title: string | null
  source_type: 'upload' | 'video_upload' | 'douyin'
  source_name: string | null
  source_url: string | null
  source_author: string | null
  author_name: string | null
  author_id: number | null
  author_uid: string | null
  play_count: number
  sort_order: number
  is_featured: boolean
  created_at: string
  updated_at: string
}

export interface MusicTrackCharacter {
  id: number
  slug: string
  name: string
  avatar_url: string | null
  theme_color: string | null
}

export interface MusicTrackDetail extends MusicTrack {
  character: MusicTrackCharacter | null
}

File: backend/app/schemas/music_track.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MusicTrackCharacterRead(BaseModel):
    id: int
    slug: str
    name: str
    avatar_url: str | None = None
    theme_color: str | None = None

    model_config = ConfigDict(from_attributes=True)


class MusicTrackRead(BaseModel):
    id: int
    slug: str
    title: str
    description: str | None = None
    audio_url: str
    cover_url: str | None = None
    duration_seconds: int | None = None
    character_id: int | None = None
    character_ids: list[int] = []
    original_title: str | None = None
    source_name: str | None = None
    source_url: str | None = None
    author_name: str | None = None
    author_id: int | None = None
    author_uid: str | None = None
    play_count: int
    sort_order: int
    is_featured: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class MusicTrackDetailRead(MusicTrackRead):
    character: MusicTrackCharacterRead | None = None
